docs/shared/guides/nx-and-ts.md

```diff
@@ -174,40 +166,45 @@ This adds a new `publish` target to our existing `project.json` configuration of
     "test": { ... },
     "publish": {
       "executor": "@nrwl/workspace:run-commands",
-      "outputs": [],
       "options": {
-        "command": "npm publish --tag=latest --access public",
-        "cwd": "dist/packages/hello-tsc"
-      }
+        "command": "node tools/scripts/publish.mjs hello-tsc {args.ver} {args.tag}"
+      },
+      "dependsOn": [{ "projects": "self", "target": "build" }]
     }
   },
   "tags": []
 }
 ```
 
-With that, we're all set up. By running the following commands we can publish our package to NPM.
+The `publish` target invokes the generated `publish.mjs` script using [`@nrwl/workspace:run-commands`](/executors/run-commands-builder) executor. The publish script does the following:
 
-```bash
-nx build hello-tsc
-nx publish hello-tsc
+- Validate the `ver` argument against a simple [SemVer](https://semver.org/) RegExp.
+- Validate the `name` of the project (eg: `hello-tsc`) against the workspace's existing projects.
+- Update the `version` property in the `package.json` of your project's `build.outputPath`
+- Invoke `npm publish` with the provided tag (default to `next` so you won't publish to `latest` by accident)
+
+Make sure to authenticate with `npm` before running the `publish` target.
+
+```shell
+nx publish hello-tsc --ver=<required-version> --tag=[custom-tag]
 ```
 
-We can automate this even further by leveraging the `targetDependencies` feature. There's clearly a dependency between `build` and `publish`, in that we want to make sure the `build` happens before the `publish` command. We can define this relationship in the `nx.json` file:
+Thanks to [“Target Dependencies” (`dependsOn`)](/configuration/projectjson#dependson) property under the `publish` target, Nx runs the `build` target automatically before Nx runs `publish`. And of course, if `build` has already run, it won't execute again, thanks to [Nx computation caching](/using-nx/caching).
 
-```json
-{
-  ...
-  "targetDependencies": {
-    ...
-    "publish": [
-      {
-        "target": "build",
-        "projects": "self"
-      }
-    ]
-  },
-  ...
-}
+> The generated `publish.mjs` script is a minimal version of what a publishing process looks like. You can definitely add more to it as you see fit for your project. For example: you can add a capability to automatic increment the version
+
+### Manual setup
```

Comment:
Does the library required to have a package.json as well if they didn't use the publish flag?

Reply:
buildable library also has `package.json` generated